### PR TITLE
[azure-cli] - python3.12 - feature broken due to distutils module removed - code fix

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Azure-cli
 
 **Description**:
 
 This PR fixes the following issue:
 
 * `Azure-cli` feature was broken due to `distutils` not provided as default in Python3.12 [#899](https://github.com/devcontainers/features/issues/899)
 
 _Changelog_:
 
 * Updated `install.sh` to add support for `bookworm`
 * Updated tests to validate that the issue was remediated
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/azure-cli/install.sh [hunk 1]
```diff
@@ -17,8 +17,8 @@ AZ_EXTENSIONS=${EXTENSIONS}
 AZ_INSTALLBICEP=${INSTALLBICEP:-false}
 INSTALL_USING_PYTHON=${INSTALL_USING_PYTHON:-true}
```
Comment (samruddhikhandale) [inline src/azure-cli/install.sh:18]: Can we bump the patch version for this Feature?

File: test/azure-cli/scenarios.json [hunk 2]
```diff
@@ -38,5 +38,15 @@
...
       }
     }
+  },
+  "install_with_python_3_12_bookworm": {
+    "image": "mcr.microsoft.com/devcontainers/python:1-3.12-bookworm",
+    "user": "vscode",
+    "features": {
+      "azure-cli": {
+        "version": "latest",
+        "installUsingPython": true
```
Comment (samruddhikhandale) [inline test/azure-cli/scenarios.json:48]: Can we add another scenario to reflect the use case in https://github.com/devcontainers/features/issues/899 ? ie with `installUsingPython: false`. (Else, you could also add `"azure-cli": {}` to the `features` object.

Comment (samruddhikhandale) [inline test/azure-cli/scenarios.json:48]: Currently, this scenario should fail as `az` does not work when installed with python (that's the issue we are trying to solve here). However, looking at the code in [here](https://github.com/devcontainers/features/blob/26883bd6a77a4f5b14a5012f884803eb0c81a766/src/azure-cli/install.sh#L191-L195), it still installs with `apt` and doesn't attempt to install via `python`. This is definitely a conflict between the experience vs the Feature option. 

I understand that this bug is not due to your PR change, however, can you fix the code? If user wants to explicitly install with python then it should not be installed with `apt`.

Comment (samruddhikhandale) [reply test/azure-cli/scenarios.json:48]: Once you fix the code, this scenario would start failing. Hence, would be good to update the scenario with `"installUsingPython": false`

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: test/azure-cli/scenarios.json (hunk 2 above)
Comment (samruddhikhandale) [reply test/azure-cli/scenarios.json:48]: Defering this to another PR

Comment (samruddhikhandale) [inline test/azure-cli/scenarios.json:48]:
```suggestion
```

File: src/azure-cli/install.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/azure-cli/install.sh]: done

File: test/azure-cli/scenarios.json (hunk 2 above)
Comment (gauravsaini04) [reply test/azure-cli/scenarios.json]: removed

**Review by samruddhikhandale — APPROVED** [review]

**Review by hhjmmmmmmmm — APPROVED** [review]

**Pull request merged**